Tasks/DeployVisualStudioTestAgent/TestAgentInstall.ps1

```diff
@@ -37,6 +37,42 @@ function Install-Product($SetupPath, $UserName, $Password, $ProductVersion, $Arg
 			Write-Warning -Verbose "Caught exception while installing Test Agent"
 			throw $_.Exception
 		}
+                 
+		if($exitCode -eq -2147185721)
+		{
+			# pending restart .
+			try
+			{
+				$testAgentFile = "$env:SystemDrive\TestAgent\testagent.txt"
+				$testAgentFileExists = Test-Path $testAgentFile
+				if($testAgentFileExists)
+				{
+					# delete the file which indicated that test agent installation failed.
+					remove-item $testAgentFile | Out-Null
+					# we have retried once .Now fail with appropriate message
+					Write-Warning -Verbose "Retried to install Test Agent"
```

Comment:
can you explain why would we provide a warning and then fail? warnings are ignored.. but here we are failing immediately post it